pep517/_in_process.py

```diff
@@ -37,6 +55,14 @@ def _build_backend():
         obj = import_module(mod_path)
     except ImportError:
         raise BackendUnavailable(traceback.format_exc())
+
+    if extra_path:
+        for path in extra_path:
+            if contained_in(obj.__file__, path):
+                break
+        else:
+            raise BackendInvalid("Backend was not loaded from backend-path")
```

Comment:
I think this check could be more concise with `if not any(...)`

Reply:
Good idea - will fix.